cpp/src/lance/arrow/utils.cc

```diff
@@ -97,18 +112,17 @@ ::arrow::Result<std::shared_ptr<::arrow::StructArray>> MergeStructArrays(
     arrays.emplace_back(left_arr);
   }
 
-  for (auto& field : rhs->struct_type()->fields()) {
-    auto& name = field->name();
-    if (lhs->GetFieldByName(name)) {
-      // We've seen each other before
-      continue;
-    }
+  for (auto name :
+       rhs->struct_type()->fields()                                                      //
+           | views::filter([&lhs](auto& f) { return !lhs->GetFieldByName(f->name()); })  //
```

Comment:
how do you feel about functional C++?

Reply:
i wish it is simpler like rust.  

It simplifies a few for for loops, but not all , 😮‍💨